### ISSUE TITLE
Support for Grouping Sets

### ISSUE THREAD

**Issue by robd003** [crate/crate#9632]
**Use case**:
This would let users do better analysis over any kind of hierarchical data generating sub-totals and a grand-total without having to run multiple queries.

Example:
SELECT SUM(plays), city, state, country FROM events GROUP BY GROUPING SETS ((city,state,country),(state,country),(country))

**Feature description**:
https://www.postgresql.org/docs/10/queries-table-expressions.html#QUERIES-GROUPING-SETS

**Comment by helenap** [comment]
@mfussenegger is this on the roadmap at all? I'd also like this feature.

**Comment by mfussenegger** [comment]
> @mfussenegger is this on the roadmap at all? I'd also like this feature.

Yes, we want to include this feature eventually, but we have no concrete plans yet.

Note that you can accomplish the same using a combination of UNION ALL - though it will be slower than a native GROUPING SETS query would be.

You could upvote the feature request here, we take the upvotes into consideration when we decide what functionality to build next.
